### PR TITLE
LibWeb: Fix crash in DOMImplementation for null namespaces and actually create XML documents

### PR DESCRIPTION
**Pull request by shannonbooth**
I was playing around again trying to figure out why attributes are not having namespaces applied, and stumbled across these small fixes (and a crash).

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp [hunk 1]
```diff
@@ -48,8 +48,8 @@ void DOMImplementation::visit_edges(Cell::Visitor& visitor)
 // https://dom.spec.whatwg.org/#dom-domimplementation-createdocument
 WebIDL::ExceptionOr<JS::NonnullGCPtr<Document>> DOMImplementation::create_document(Optional<String> const& namespace_, String const& qualified_name, JS::GCPtr<DocumentType> doctype) const
 {
-    // FIXME: 1. Let document be a new XMLDocument
-    auto xml_document = Document::create(realm());
+    // 1. Let document be a new XMLDocument
+    auto xml_document = XMLDocument::create(realm());
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp:52]: Typo in commit message: this is createDocument, not createElement :^)

File: Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Document>> DOMImplementation::create_docume
 
     // 3. If qualifiedName is not the empty string, then set element to the result of running the internal createElementNS steps, given document, namespace, qualifiedName, and an empty dictionary.
     if (!qualified_name.is_empty())
-        element = TRY(xml_document->create_element_ns(namespace_.value(), qualified_name, ElementCreationOptions {}));
+        element = TRY(xml_document->create_element_ns(namespace_, qualified_name, ElementCreationOptions {}));
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp:61]: Since this fixes a crash, can we add a test?

Comment (shannonbooth) [reply Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp:61]: Yup, that's actually covered in the test added in the following commit

File: Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp (hunk 1 above)
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/DOM/DOMImplementation.cpp:52]: derp, thanks!

**Pull request merged**